flink-python/pyflink/common/restart_strategy.py

```diff
@@ -0,0 +1,244 @@
+
+        def get_delay_between_attempts_interval(self):
+            return from_j_flink_time(self._j_restart_strategy_configuration
+                                     .getDelayBetweenAttemptsInterval())
+
+    class FallbackRestartStrategyConfiguration(RestartStrategyConfiguration):
+        """
+        Restart strategy configuration that could be used by jobs to use cluster level restart
+        strategy. Useful especially when one has a custom implementation of restart strategy set via
+        flink-conf.yaml.
+        """
+
+        def __init__(self, j_restart_strategy=None):
+            if j_restart_strategy is None:
+                gateway = get_gateway()
+                self._j_restart_strategy_configuration = \
+                    gateway.jvm.RestartStrategies.FallbackRestartStrategyConfiguration()
+                super(RestartStrategies.FallbackRestartStrategyConfiguration, self)\
+                    .__init__(self._j_restart_strategy_configuration)
+            else:
+                super(RestartStrategies.FallbackRestartStrategyConfiguration, self)\
+                    .__init__(j_restart_strategy)
+
+    @classmethod
+    def _from_j_restart_strategy(cls, j_restart_strategy):
+        if j_restart_strategy is None:
+            return None
+        gateway = get_gateway()
+        NoRestartStrategyConfiguration = gateway.jvm.RestartStrategies\
+            .NoRestartStrategyConfiguration
+        FixedDelayRestartStrategyConfiguration = gateway.jvm.RestartStrategies\
+            .FixedDelayRestartStrategyConfiguration
+        FailureRateRestartStrategyConfiguration = gateway.jvm.RestartStrategies\
+            .FailureRateRestartStrategyConfiguration
+        FallbackRestartStrategyConfiguration = gateway.jvm.RestartStrategies\
+            .FallbackRestartStrategyConfiguration
+        clz = j_restart_strategy.getClass()
+        if clz.getName() == get_java_class(NoRestartStrategyConfiguration).getName():
+            return RestartStrategies.NoRestartStrategyConfiguration(
+                j_restart_strategy=j_restart_strategy)
+        elif clz.getName() == get_java_class(FixedDelayRestartStrategyConfiguration).getName():
+            return RestartStrategies.FixedDelayRestartStrategyConfiguration(
+                j_restart_strategy=j_restart_strategy)
+        elif clz.getName() == get_java_class(FailureRateRestartStrategyConfiguration).getName():
+            return RestartStrategies.FailureRateRestartStrategyConfiguration(
+                j_restart_strategy=j_restart_strategy)
+        elif clz.getName() == get_java_class(FallbackRestartStrategyConfiguration).getName():
+            return RestartStrategies.FallbackRestartStrategyConfiguration(
+                j_restart_strategy=j_restart_strategy)
+        else:
+            raise Exception("Unsupported java RestartStrategyConfiguration: %s" % clz.getName())
+
+    @classmethod
```

Comment:
Seems that we can declare these kinds of methods static. Should also check methods in other places such as CheckpointingMode, etc.